server.h

```diff
@@ -278,8 +297,50 @@ class CTFOServer {
   }
 
   void OnMidichloriansEvent(const LogEntry& entry) {
-    // TODO(mzhurovich): update answers here.
-    static_cast<void>(entry);
+    std::unique_ptr<MidichloriansEvent> event;
+    if (entry.m == "POST") {
+      try {
+        ParseJSON(entry.b, event);
+        UpdateStateOnEvent(event);
+      } catch (const bricks::ParseJSONException&) {
+        // TODO(mzhurovich): Error logging.
+        std::cerr << "Parse JSON failed." << std::endl;
+      }
+    }
+  }
+
+  void UpdateStateOnEvent(const std::unique_ptr<MidichloriansEvent>& event) {
+    try {
+      const iOSGenericEvent& ge = dynamic_cast<const iOSGenericEvent&>(*event.get());
```

Comment:
I don't get it.
```
[C++11: 5.2.7/9]: The value of a failed cast to pointer type is the null pointer value of the required result type.
A failed cast to reference type throws std::bad_cast (18.7.2).
```
I try to cast to reference, so I expect `std::bad_cast` on failure.